warehouse/http.py

```diff
@@ -0,0 +1,34 @@
+import threading
+import requests
+
+
+_local = threading.local()
```

Comment:
I think it would be better to have this be encapsulated inside of the ``ThreadLocalSessionFactory`` instead of having it be a global thread local. I'd just move this line into ``ThreadLocalSessionFactory.__init__``. That should still give thread local sessions, but have separate pools per instance of ``ThreadLocalSessionFactory``, which by default we only create one of, but there's no reason we couldn't do more.